File: MatlabOctaveCodeSnippets/snip_signals_data_precision.m
str = 'Ranges for double before and after 0:\n%g to %g and %g to %g';
sprintf(str, -realmax, -realmin, realmin, realmax)
str = 'Ranges for float before and after 0:\n%g to %g and %g to %g';
sprintf(str,-realmax('single'),-realmin('single'), ...
		realmin('single'), realmax('single'))

File: PythonCodeSnippets/snip_signals_data_precision.py
import numpy as np

fmax = np.finfo(np.float64).max
fmin = np.finfo(np.float64).min
nfmax = -fmax
nfmin = -fmin
print(
    f"Ranges  for  double  before  and  after  0:\n{nfmax} to {nfmin} and {fmin} to {fmax}"
)

fmax = np.finfo(np.float32).max
fmin = np.finfo(np.float32).min
nfmax = -fmax
nfmin = -fmin
print(
    f"Ranges  for  float  before  and  after  0:\n{nfmax} to {nfmin} and {fmin} to {fmax}"
)

print(np.finfo(0.135).eps)
